.github/workflows/robot/internal/bot/backport.go

```diff
@@ -0,0 +1,203 @@
+	"strings"
+	"text/template"
+
+	"github.com/gravitational/trace"
+)
+
+func (b *Bot) Backport(ctx context.Context) error {
+	if !b.c.Review.IsInternal(b.c.Environment.Author) {
+		return trace.BadParameter("automatic backports are only supported for internal contributors")
+	}
+
+	pull, err := b.c.GitHub.GetPullRequest(ctx,
+		b.c.Environment.Organization,
+		b.c.Environment.Repository,
+		b.c.Environment.Number)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	branches := findBranches(pull.UnsafeLabels)
+
+	var statuses []Status
+	for _, base := range branches {
+		status := "Success"
+		number, err := b.backportBranch(ctx,
+			b.c.Environment.Organization,
+			b.c.Environment.Repository,
+			b.c.Environment.Number,
+			pull.UnsafeTitle,
+			base)
+		if err != nil {
+			status = "Failure"
+			log.Printf("Failed to backport %v to %v: %v.", b.c.Environment.Number, base, err)
+		}
+
+		statuses = append(statuses, Status{
+			Status: status,
+			Branch: base,
+			Link: url.URL{
+				Scheme: "https",
+				Host:   "github.com",
+				Path:   path.Join(b.c.Environment.Organization, b.c.Environment.Repository, "pull", strconv.Itoa(number)),
+			},
+		})
+	}
+
+	err = b.updatePullRequest(ctx,
+		b.c.Environment.Organization,
+		b.c.Environment.Repository,
+		b.c.Environment.Number,
+		statuses)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+// findBranches looks through the labels attached to a Pull Request for all the
+// backport branches the user requested.
+func findBranches(labels []string) []string {
+	var branches []string
+
+	for _, label := range labels {
+		if !strings.HasPrefix(label, "backport/") {
+			continue
+		}
+
+		branches = append(branches, strings.TrimPrefix(label, "backport/"))
+	}
+
+	return branches
+}
+
+// TODO(russjones): Manually test backport logic to make sure it is robot in
+// the case of failure to cherry-pick a commit.
+func (b *Bot) backportBranch(ctx context.Context, organization string, repository string, number int, title string, base string) (int, error) {
+	git("config", "--global", "user.email", "bot@goteleport.com")
+	git("config", "--global", "user.name", "github-actions")
+
+	head := fmt.Sprintf("bot/backport-%v", number)
+
+	// Fetch base and head branches and create the backport branch that tracks
+	// the branch the Pull Request will be backported to.
+	// TODO(russjones): Check for injection attacks here.
+	if err := git("fetch", "origin", base); err != nil {
+		return 0, trace.Wrap(err)
+	}
+	if err := git("fetch", "origin", b.c.Environment.UnsafeHead); err != nil {
+		return 0, trace.Wrap(err)
+	}
+	if err := git("checkout", "-b", head, "--track", fmt.Sprintf("origin/%v", base)); err != nil {
+		return 0, trace.Wrap(err)
+	}
+
+	// Get list of commits to backport and cherry-pick to backport branch.
+	commits, err := b.c.GitHub.ListCommits(ctx,
+		organization,
+		repository,
+		number)
+	if err != nil {
+		return 0, trace.Wrap(err)
+	}
+	for _, commit := range commits {
```

Comment:
Isn't this a more convoluted way to do `git cherry-pick origin/basebranch..origin/prbranch`? Why do we need to loop ourselves? Also, won't this pick up merge commits along the way? (which the ranged form of cherry-pick won't do, I believe)

Reply:
Huh, didn't know you could do ranges like this with `git cherry-pick`, will update.

Reply:
Cherry picking a range fails the same way cherry picking a single commit. However `git rebase -onto ...` does work and should drop merge commits. It's the third option here: https://stackoverflow.com/questions/29914052/how-to-git-rebase-a-branch-with-the-onto-command